### ISSUE TITLE
Square brackets shown in preview of wikilinks with reference definitions

### ISSUE THREAD

**Issue by josephdecock** [foambubble/foam#960]
### Describe the bug

Wikilinks are displayed with one level of outer square brackets in preview if there is a Link Reference Definition and not an alias in the link itself. Wikilinks without a Link Reference Definition are displayed without any square brackets, as are wikilinks with an alias.

### Small Reproducible Example

https://github.com/josephdecock/foam-reference-link-bug

### Steps to Reproduce the Bug or Issue

1. Clone https://github.com/josephdecock/foam-reference-link-bug
2. Open markdown preview of readme.md

### Expected behavior

All wikilinks should be previewed without square brackets.

### Screenshots or Videos

![image](https://user-images.githubusercontent.com/1145533/160242290-de39ff31-6eb3-4377-9248-9665192a02b9.png)

### Operating System Version

Windows

### Visual Studio Code Version

1.65.2

**Comment by josephdecock** [comment]
Hey there, just wanted to check in on this issue and see if there's any additional info needed for it. Thanks very much!

**Comment by riccardoferretti** [comment]
Hi @josephdecock, this is not high up on the priority list, but I agree the current behavior is sub-optimal, would be happy to support someone looking into this (for reference, the place to start would be `markdownItWithFoamLinks` in `packages/foam-vscode/src/features/preview-navigation.ts`

**Comment by josephdecock** [comment]
Thanks, I took a look at that code and what I've observed in the debugger is that when the plugin created by markdownItWithFoamLinks runs, it gets no matches on the ref links. It looks markdownIt has already replaced the reference with an `<a>` tag at that point. A thing that I've tried that seems to work is to remove all references that are formatted like wikilinks, similar to how references are removed to keep them from breaking note inclusion. 

If you like this approach, I can clean it up and submit a PR. I'm thinking probably we don't need a special case for removing references with the alias separator, the comments need to be updated, and a few tests would be good too.
